README.md

```diff
@@ -0,0 +1,69 @@
+# Educational-website-and-courses-finder-for-keyword
+Return the filtered useful educational resources based on input keyword and filtering conditions from users
+## Functional Design(secondary version, subject to change)
+### Module: DataCollection
+1. Function: DataCollection
+   - Description: The search process for resources gathering
+   - Input:
+     - keywords: list of keywords for related educational resources finding
+   - Output:
+     - raw_data: unprocessed related resources found through the search, stored in lists for future processing
+   - Functionality: Obtain searching resources from Google Search with keywords
+
+2. Function: Preprocessing(For classifier)
+   - Description: Process the resources for later processing
+   - Input:
+     - raw_data: unprocessed related resources found through the search, stored in lists for future processing
+   - Output:
+     - ripe_data: processed related resources found through the search, format friendly for later modules
+   - Functionality: Turn the rough format of resources into the format and syntax that later modules want
+
+### Module: UsefulnessRanking(This should be a large module, with many incoming supplement functions)
+1. Function: ResourceRanking(Firstly implement by 'many-if')
+   - Description: Rank the usefulness of collected resources
+   - Input:
+     - ripe_data: processed related resources found through the search, format friendly for later modules
+   - Output:
+     - rank: a list that contains the ranking for resources of the same index in ripe_data accordingly
+   - Functionality: Provide the ranking of usefulness for all resources, used as judgment criteria in later module
+
+### Module: FurtherResearch
+1. Function: FurtherResearch(extra)
+   - Description: For each of the classified result, track some furthur features, include but not limited to user comments, user rankings, page view, share or reprint times, favorite perventage, release time, publisher etc..
+   - Input:
+     - ripe_data: processed related resources found through the search, format friendly for later modules
+     - (extra)conditions: filtering conditions for returned value according to user input string
+   - Output:
+     - filtered_data:  a list that contains the further ranking for resources of the same index in ripe_data accordingly
+   - Functionality: Track all top results upon certain features for further usefulness evaluation
+
+## Algorithm Design(secondary version, subject to change)
+### (New version)only naive primary thought, no reference
+1. Using a crawler to copy down the source code of html page for every individual google search result according to input keyword.</br>
+![172eee9c7336938393c2ec712c2ff97](https://user-images.githubusercontent.com/66361320/135891667-90540080-ca48-456f-bb52-f97ef9aeb3da.png)
+
+2. Judge usefulness based on the following features, every single point satisfied will add up score for that result, and different point may provide different score:</br>
```

Comment:
I suggest combining the modules `URL check`, `Title check`, and `Content check` blocks into a single block called `Classifier` in the diagram. You can keep the descriptions of features below the diagram.